### DISCUSSION TITLE
Recommended approach for defining keys on aggregate functions?

### DISCUSSION THREAD

**Discussion by GollyJer** [urql-graphql/urql#1274] [Q&A] [answered]
I'm grabbing the count from our Hasura graphql database.

```javascript
export const GET_ME = gql`
  query GET_ME {
    user {
      user_chat_groups_aggregate {
        aggregate {
          count
        }
      }
  }
`;
```

The problem I'm hitting is `user_chat_groups_aggregate` and `aggregate ` have no defining characteristics.
```javascript 
{
  "user_chat_groups_aggregate": {
    "aggregate": {
      "count": 1
    }
  }
}
```

This... 
```javascript 
keys: {
    chat_group_user_aggregate: () => null,
},
```
results in this error.
`user:2.user_chat_groups_aggregate.aggregate  has a selection set, but no key could be generated`

What is the recommended way to handle the nested result with no keys in this situation?

Thanks!

**Comment by GollyJer** [comment, marked as answer]
OK, literally two minutes after posting this I read the error more clearly.

It looks like this is the solution.

```javascript 
keys: {
  chat_group_user_aggregate: () => null,
  chat_group_user_aggregate_fields: () => null,
},
```

No more warnings but **is this correct**? Thanks! 👍

**Reply by kitten** [reply]
Yes, it seems correct as long as this aggregate type is supposed to be embedded, meaning it'll only be accessed at this place in the schema/query (only matters in relation to its parent), has no keyable field, or isn't shared with any other part of another query ✌️
